components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHost.java

```diff
@@ -85,7 +85,7 @@ public synchronized void registerHandler(HttpHandlerRegistrationInfo registratio
 
         String path = registrationInfo.getUri().getPath();
         String methods = registrationInfo.getMethodRestrict();
-        boolean prefixMatch = registrationInfo.isMatchOnUriPrefix();
+        boolean prefixMatch = !registrationInfo.isMatchOnUriPrefix();
```

Comment:
Yeah that can't be right. The original code should be fine here (same for the other similar change below).

With this change `prefixMatch` will equal `true` if  the `matchOnUriPrefix` endpoint config value is `false`. Which is the total opposite of the desired behaviour. 

We only want the Undertow consumer to match on URI prefixes if `matchOnUriPrefix == true`.